gtsam/base/OptionalJacobian.h

```diff
@@ -66,11 +61,18 @@ class OptionalJacobian {
 
 public:
 
-  /// Default constructor acts like boost::none
+  /// Default constructor
   OptionalJacobian() :
       map_(nullptr) {
   }
 
+  /// Default constructor with nullptr_t
```

Comment:
Why was this not needed before?

Reply:
Because in this PR I replaced all `OptionalJacobian(boost::none)` to `OptionalJacobian(nullptr)`.

Reply:
I think nullptr was already accepted as an argument, but maybe it was silently cast to boost::none?

Reply:
Possible that's what happend